megatron/training/arguments.py

```diff
@@ -900,7 +900,7 @@ def validate_args(args, defaults={}):
             'residual connection in fp32 only supported when using fp16 or bf16.'
 
     if args.moe_grouped_gemm:
-        assert args.bf16, 'Currently GroupedGEMM for MoE only supports bf16 dtype.'
+        assert args.fp16 or args.bf16, 'Currently GroupedGEMM for MoE only supports fp16 or bf16 dtype.'
```

Comment:
This assertion is for the legacy grouped gemm https://github.com/NVIDIA/Megatron-LM/blob/299034c2f31c3b689f6baba6c3da79e5d16e6cc3/megatron/core/transformer/moe/experts.py#L102. We should move this assertion to that module.